_test/common_functions/is_cut_equal.m

```diff
@@ -12,48 +14,85 @@
         else
-            ok = false;
-            mess=sprintf(['Pixels parameters of two cuts are different:\n',...
-                '   npix1=%d and npix2 = %d\n',...
-                'pix_range1 = [%f  %f %f %f; pix_range2=[%f  %f %f  %f;\n',...
-                '              %f  %f %f %f]             %f  %f %f  %f]'],...
-                w1tot.data.pix.num_pixels,w2tot.data.pix.pix_range,...
-                w1tot.data.pix.pix_range(1,:),w2tot.data.pix.pix_range(1,:),...
-                w1tot.data.pix.pix_range(2,:),w2tot.data.pix.pix_range(2,:));
+            if equal_to_tol(w1tot.data.pix.pix_range,w2tot.data.pix.pix_range,tol) && ...
+                    equal_to_tol(w1tot.data.pix.num_pixels,w2tot.data.pix.num_pixels,tol)
+                ok=true;
+                mess='';
+            else
+                ok = false;
+                mess=sprintf(['Pixels parameters of two cuts are different:\n',...
+                              '   npix1=%d and npix2 = %d\n',...
+                              'pix_range1 = [%f  %f %f %f; pix_range2=[%f  %f %f  %f;\n',...
+                              '              %f  %f %f %f]             %f  %f %f  %f]'],...
+                             w1tot.data.pix.num_pixels,w2tot.data.pix.pix_range,...
+                             w1tot.data.pix.pix_range(1,:),w2tot.data.pix.pix_range(1,:),...
+                             w1tot.data.pix.pix_range(2,:),w2tot.data.pix.pix_range(2,:));
+            end
         end
+    else
+        ok=false;
+        mess='One or more of npix, s, e are not the same';
+    end
+end
+
+function [f1, f2, proj, pN, cutArgs, tol] = parse_args(varargin)
+    p = inputParser();
+    addRequired(p, 'f1', @validate_sqw);
+    addRequired(p, 'f2', @validate_sqw);
+    addRequired(p, 'proj', @validate_proj);
+    addRequired(p, 'p1',     @(x)(validateattributes(x,{'numeric'},{'nonempty','vector'})));
+    addRequired(p, 'p2',     @(x)(validateattributes(x,{'numeric'},{'nonempty','vector'})));
+    addOptional(p, 'p3', [], @(x)(validateattributes(x,{'numeric'},{'vector'})));
+    addOptional(p, 'p4', [], @(x)(validateattributes(x,{'numeric'},{'vector'})));
+    addParameter(p, 'cutArgs', {}, @iscell)
+    addParameter(p, 'tol', [0,0], @(x)(validateattributes(x,{'numeric'},{'nonnegative','vector'})))
+    parse(p, varargin{:});
+
+    f1 = p.Results.f1;
+    f2 = p.Results.f2;
+    if ~iscell(f1), f1={f1}; end
+    if ~iscell(f2), f2={f2}; end
+
+    proj = p.Results.proj;
+    pN = {p.Results.p1, p.Results.p2, p.Results.p3, p.Results.p4};
+    pN = pN(~cellfun(@isempty, pN));
+    tol = p.Results.tol;
+    cutArgs = p.Results.cutArgs;
+end
+
+function ok = validate_sqw(inp)
+    check = @(x)(isa(inp, 'sqw') || ischar(inp));
+    ok = check(inp) || (iscell(inp) && all(cellfun(check, inp)));
+end
+
+function ok = validate_proj(inp)
+    ok = isfield(inp, 'u') && isfield(inp, 'v');
+    if ok
+        validateattributes(inp.u, {'numeric'}, {'nonempty','vector','numel',3});
```

Comment:
if `validateattributes` fails `ok` is not set to false

Reply:
In a spout of MATLAB genius, `validateattributes` raises if it's not met, it cannot set ok. I guess, I should probably catch and set `ok` to not `ok`.